Issue-28: correctly handle modules with custom file paths

JJSwigut opened the pull request.

JJSwigut committed:
correctly handle modules with custom file paths

JJSwigut committed:
Tests and formatting

j-roskopf commented:
@JJSwigut this looks really good!

I tried it out on work and the only thing that is a slight issue is that there is an initial file separator in the beginning of the file path.

`includeProject(":foundation:test","/foundation/test")`

JJSwigut committed:
better construct text function, better tests

JJSwigut commented:
> @JJSwigut this looks really good!
> 
> I tried it out on work and the only thing that is a slight issue is that there is an initial file separator in the beginning of the file path.
> 
> `includeProject(":foundation:test","/foundation/test")`

@j-roskopf 
There was some weirdness in the `constructTextToWrite` function so I rewrote it to better handle removing the "/" prefix when necessary. I also updated to AssertEquals tests so it better catches the differences and makes debugging it easier. Should be gucci now though!

j-roskopf commented on src/main/kotlin/com/joetr/modulemaker/file/FileWriter.kt:
```diff
@@ -272,13 +274,14 @@ class FileWriter(
     ) {
         val settingsFile = Files.readAllLines(Paths.get(settingsGradleFile.toURI()))
-
+        println(rootPathAsString)
```
can we get rid of this bad lad

j-roskopf commented on src/main/kotlin/com/joetr/modulemaker/ModuleMakerDialogWrapper.kt:
```diff
@@ -587,6 +587,7 @@ class ModuleMakerDialogWrapper(
         val currentlySelectedFile = getCurrentlySelectedFile()
         if (settingsGradleFile != null) {
             fileWriter.createModule(
+                rootPathString = selectedSrcValue.split(File.separator).drop(1).joinToString(File.separator),
```
can you add a comment with this is doing with expected input?

something like

// at this point, selectedSrcValue has a value of something like `/Root/Module/Module2/` - we want to remove the root of the project to use as the file path in settings.gradle

JJSwigut committed:
remove test code, add comments
